Add GitHub Actions for Tests

eumiro opened the pull request:
Convert the Test CI from Travis to GitHub Actions.

avalentino commented:
OK, we cannot see GHA tests until GHA itself is not activated in the repo.
I suggest to merge a first PR with a dummy GHA setup, e.g. just checkout sources (more or less what you already have).
At that point we will have the  GHA enabled and we can setup a new PR with the complete implementation.
Does it make sense for you?

cc @amotl

eumiro commented:
The GHA results are [here](https://github.com/eumiro/PyTables/actions) but I am learning the syntax on the fly, so it still does not work correctly. Anyway we don't need to merge to `master` right now. If you have any ideas, how to make it work, let me know.

eumiro commented:
FYI, @avalentino I found a YAML formatting error, fixed it, and now it runs apparently for more than 20 minutes (it still did not appear in the list of builds), which sounds like it should do something now. If that fails again, I will limit the list of Python versions to a single one to fail faster.

Oh, there's a gitter.im chat if you want to follow…

avalentino commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,34 @@
+      - name: 'Run test'
+        run: |
+          pip install cython
+          pip install dist/*.tar.gz
```
IMHO before running
```
pip install dist/*.tar.gz
```
there should be something like
```
python3 setup.py sdist
```
or
```
make dist
```

eumiro replied:
The long run is still now over. I will now change that line to `python3 setup.py dist` and limit Python versions to 3.5.

avalentino commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,34 @@
+        uses: actions/setup-python@v2
+        with:
+          python-version: '${{ matrix.python-version }}'
+      # - run: sudo apt-get install libbz2-dev libhdf5-serial-dev liblzo2-dev
```
Without this PyTables is not going to build

eumiro replied:
Thanks, I added that line to the long list of `run` in `Run test`.

avalentino commented:
> If that fails again, I will limit the list of Python versions to a single one to fail faster.

I agree, it would accelerate a lot things for the initial development.

eumiro commented:
Did I break GitHub? Even after more than an hour there is no response and adding new workflow files, even very basic, does not trigger any new builds. :man_shrugging:

avalentino commented:
> Did I break GitHub? Even after more than an hour there is no response and adding new workflow files, even very basic, does not trigger any new builds.

Indeed there is something strange.
Please let go the CI on this dummy workflow and I will merge it.
After that we shall be able to see GHA stuff also in the PR and on the main repo.

avalentino commented:
@eumiro  I can confirm that also on my personal fork I'm not able to start GHA, there must be some problem on GH side.
I will try again later.

eumiro commented:
> > Did I break GitHub? Even after more than an hour there is no response and adding new workflow files, even very basic, does not trigger any new builds.
> 
> Indeed there is something strange.
> Please let go the CI on this dummy workflow and I will merge it.
> After that we shall be able to see GHA stuff also in the PR and on the main repo.

If you don't want to pollute `master` with our attempts, create a new branch on the main repository and we'll define the GHA against this branch. Then we can develop the GHA configuration and PR it against it as well. When everything works, we'll repeat it at once with `master`.

I'll wait now until there's a working dummy GHA on the main repository.

eumiro commented:
> @eumiro OK, please rebase on the current master

Rebased and I updated the `ci.yml` file. There is still a problem with the tests, but at least they get launched, even in a PR!

avalentino commented on .github/workflows/ci.yml:
```diff
@@ -1,6 +1,12 @@
+      - master
+  pull_request:
+    branches:
+      - master
```
Not sure but I think that this one is the cause of GHA not triggered on PR (not matching "master").
Probably using the following:
```
on: [push pull_request]
```
is OK.
After all we do not have an high commit rate.

avalentino commented:
@eumiro when you are fine with your changes please remember to update the PR title and description.

avalentino commented:
Fantastic!
Thanks a lot @eumiro
